qiskit/transpiler/passes/routing/star_prerouting.py

```diff
@@ -288,10 +166,26 @@ def determine_star_blocks_processing(
             List[StarBlock]: a list of star blocks in the given dag
             Union[List[DAGOpNode], List[DAGDepNode]]: a list of operations specifying processing order
         """
-        blocks, processing_order = self.collect_all_matching_blocks(
-            dag, min_block_size=min_block_size
+
+        def filter_fn(node):
+            """Specifies which nodes can be collected into star blocks."""
+            return (
+                len(node.qargs) <= 2
+                and len(node.cargs) == 0
+                and getattr(node.op, "condition", None) is None
+                and not isinstance(node.op, Barrier)
+            )
+
+        block_collector = BlockCollector(dag)
+        matching_blocks = block_collector.collect_all_matching_blocks(
+            filter_fn=filter_fn,
+            block_class=StarBlock,
+            output_nodes=False,
+            min_block_size=min_block_size,
```

Comment:
Oh I see, we do not catch the `DAGCircuitError` here as the `split` method for `StarBlock` always returns `[self]`, which is not `None`. Still we should have `split_blocks=False` for this section to avoid confusion and unnecessary compute.

Reply:
Yeah, I agree, the previous handling was sloppy, as you have pointed out.